bson/codec_options.py

```diff
@@ -485,7 +485,7 @@ def with_options(self, **kwargs: Any) -> "CodecOptions":
             return CodecOptions(**opts)
 
 
-DEFAULT_CODEC_OPTIONS: "CodecOptions[Mapping[str, Any]]" = CodecOptions()
```

Comment:
I think all of the places you changed to "Dict" can be replaced with "Mapping", if you use a cast in `_drop_helper` to convert the result from `_command` to `Dict`.

Reply:
I agree that would work. Why is `Mapping` preferred over `Dict`? When looking into the `CodecOptions` class, it looks like the default is a dict.

Reply:
In general we want to use read-only interfaces where possible.  The underlying implementation is a `dict`, but it should be considered read-only for most use cases.

Reply:
I think we should prefer to return Dict when we actually do return Dicts. Like db.command("ping") ideally returns a Dict, since that's what it actually returns.

Reply:
I guess it comes down to whether we want to expose the fact that the return value is mutable.  In the case of `command`, I agree, we could expose this fact.

Reply:
Originally, I changed them to `Dict` because I thought it would be better to use the more specific type, especially when we know for sure that it is a `Dict`